### PR TITLE
std: Rework `std.math.order` and `std.math.Order`

### PR DESCRIPTION
**Pull request by SeanTheGleaming**
Revival of #23291
- std.math.{order, Order} are now implemented in their own file (std/math/order.zig)
- The values of `Order` are now explicitly specified
  - To be specific, the values are as follows:
    -  `lt = -1`
    - `eq = 0`
    - `gt = 1`
  - Leveraging this enabled some optimizations for `Order.invert` and `Order.compare`
- `std.math.order` is now implemented branchlessly, for both floats and integers
- `std.math.order` for floats now uses IEEE total ordering
  - That is to say, numbers are ordered as follows: `{-nan, -inf, -real, -0, +0, +real, inf, nan}`
  - Previously, ordering nans would result in illegal behavior. Now, this is well defined.
  - This is not a breaking change, as it only affects cases which were previously illegal
- Additionally fixed `std.math.big.int.subCarry` returning `-0` when subtracting `0 - 0`. This had in turn caused some compiler bugs with comptime arithmetic, which made themselves apparent while working on this
  - This bug was why I closed the original PR

### CODE REVIEW

File: lib/std/math/order.zig [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    try expectEqual(.eq, order(0, 0));
+
+    // -1 < 1
+    try expectEqual(.lt, order(-1, 1));
+
+    // 1 > -1
+    try expectEqual(.gt, order(1, -1));
+}
+
+test "total ordering" {
```
Comment (castholm) [inline lib/std/math/order.zig:119]: For completeness and just to make sure that all the edge cases are covered, it might be nice to have tests cases that compare qNaN and sNaN (IEEE 754 defines totalOrder such that -qNaN < -sNaN and +sNaN > +qNaN), as well as cases that assert two different NaNs of the same kind are correctly ordered by payload.

Comment (SeanTheGleaming) [reply lib/std/math/order.zig:119]: I have addressed this on my local copy, but I'm going to hold off on pushing it until I finish wrangling some failed behavior tests and bugs. In any case, I will be including this

**Comment by SeanTheGleaming** [comment]
As is indicated by the repeated failure of tests, this change is becoming much more involved than it seemed at a glance. I expect to have this finished in a few days.

**Comment by SeanTheGleaming** [comment]
Now that this is (finally, haha) done and in working order, I would like to hear any opinions on the breakage made by total ordering. This only changes how zeros of different signs are ordered, so it is a niche case, but it could be worth considering. Despite that, I think total ordering is the sanest way to go about this; it is a part of the IEEE 754 standard and covers all edge cases.

If we want to eliminate breakage, we could split the current `order` into `order` and `totalOrder`, but that would beg the question of how `order` would handle nan values, as frankly, illegal behavior doesn't seem like a tenable option here, at least in my opinion (we don't want to cause panics anytime someone tries to do a simple floating point comparison!).
